streams/src/main/java/org/apache/kafka/streams/processor/internals/TaskManager.java

```diff
@@ -1140,4 +1166,12 @@ public static void executeAndMaybeSwallow(final boolean clean,
             throw e; },
             e -> log.debug("Ignoring error in unclean {}", name));
     }
+
+    boolean hasPreRunningTasks() {
+        return tasks().values().stream().anyMatch(Task::preRunning);
+    }
+
+    public void setResetStrategy(final Function<TopicPartition, OffsetResetStrategy> resetStrategy) {
```

Comment:
I think the name of the method is not ideal. We don't set the strategy, but we set a function that can compute the strategy for a partitions. Needed to go forth and back on the PR to understand how it work, and assume the method name had its part in confusing me.

Reply:
Sorry about that. I was only thinking "set a field named 'reset strategy'", but now I see that it sounds more like what you said. I'll rename it.